test/controllers/CustomsDutyControllerSpec.scala

```diff
@@ -0,0 +1,134 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.ControllerSpecBase
+import controllers.Assets.Redirect
+import controllers.actions.FakeDataRetrievalAction
+import forms.CustomsDutyFormProvider
+import mocks.repositories.MockSessionRepository
+import models.{UnderpaymentAmount, UnderpaymentType, UserAnswers}
+import pages.{CustomsDutyPage, UnderpaymentTypePage}
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsFormUrlEncoded, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{charset, contentType, defaultAwaitTimeout, status}
+import views.html.CustomsDutyView
+
+import scala.concurrent.Future
+
+class CustomsDutyControllerSpec extends ControllerSpecBase {
+
+  val userAnswersWithUnderpayment = Some(UserAnswers("some-cred-id")
+    .set(
+      UnderpaymentTypePage,
+      UnderpaymentType(true, false, false)
+    ).success.value
+  )
+
+  private def fakeRequestGenerator(original: String, amended: String): FakeRequest[AnyContentAsFormUrlEncoded] =
+    fakeRequest.withFormUrlEncodedBody(
+      "original" -> original,
+      "amended" -> amended
+    )
+
+  trait Test extends MockSessionRepository {
+    lazy val controller = new CustomsDutyController(
+      authenticatedAction,
+      dataRetrievalAction,
+      dataRequiredAction,
+      mockSessionRepository,
+      messagesControllerComponents,
+      customsDutyView,
+      form
+    )
+    private lazy val customsDutyView = app.injector.instanceOf[CustomsDutyView]
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+    val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id"))
+    val formProvider: CustomsDutyFormProvider = injector.instanceOf[CustomsDutyFormProvider]
+    MockedSessionRepository.set(Future.successful(true))
+    val form: CustomsDutyFormProvider = formProvider
+  }
+
+  "GET /" should {
+    "return OK" in new Test {
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "return HTML" in new Test {
+      override val userAnswers: Option[UserAnswers] = Option(
+        UserAnswers("some-cred-id").set(
+          CustomsDutyPage,
+          UnderpaymentAmount(BigDecimal("40"), BigDecimal(40))
+        ).success.value
+      )
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+      charset(result) mustBe Some("utf-8")
+    }
+
+  }
+
+  "POST /" when {
+
+    "should redirect to Import VAT page" in new Test {
+      controller.redirect(Some(UnderpaymentType(true, true, false))) mustBe Redirect(controllers.routes.CustomsDutyController.onLoad()) // Import VAT
+    }
```

Comment:
Please rewrite this and other navigation tests like this:
```suggestion
    "Import VAT was selected as an underpayment" should {
      "redirect to Import VAT page" in new Test {
        controller.redirect(Some(UnderpaymentType(true, true, false))) mustBe 
  Redirect(controllers.routes.CustomsDutyController.onLoad()) // Import VAT
        }
      }
```
Current output:
```
[info] POST /
[info] - when should redirect to Import VAT page
[info] - when should redirect to Excise Duty page
[info] - when should redirect to Summary page
[info]   when payload contains valid data
[info]   - should return a SEE OTHER response when correct data is sent
[info]   - should update the UserAnswers in session
[info]   when payload contains invalid data
[info]   - should return BAD REQUEST when original amount is exceeded
[info]   - should return BAD REQUEST
```
Expected output:
```
[info] POST /
[info]   when Import VAT was selected as an underpayment
[info]   - should redirect to Import VAT page
[info] - when should redirect to Excise Duty page
[info] - when should redirect to Summary page
```